src/util.cpp

```diff
@@ -943,18 +954,38 @@ void ArgsManager::ReadConfigStream(std::istream& stream)
     }
 }
 
-void ArgsManager::ReadConfigFile(const std::string& confPath)
+void ArgsManager::ReadConfigFiles()
 {
     {
         LOCK(cs_args);
         m_config_args.clear();
     }
 
+    const std::string confPath = GetArg("-conf", BITCOIN_CONF_FILENAME);
     fs::ifstream stream(GetConfigFile(confPath));
 
     // ok to not have a config file
     if (stream.good()) {
         ReadConfigStream(stream);
+        if (m_override_args.count("-includeconf") == 0) {
+            std::vector<std::string> includeconf(GetArgs("-includeconf"));
+            {
+                // We haven't set m_network yet (that happens in SelectParams()), so manually check
+                // for network.includeconf args.
+                std::vector<std::string> includeconf_net(GetArgs(std::string("-") + GetChainName() + ".includeconf"));
+                includeconf.insert(includeconf.end(), includeconf_net.begin(), includeconf_net.end());
+            }
+
+            for (const std::string& to_include : includeconf) {
+                fs::ifstream include_config(GetConfigFile(to_include));
+                if (include_config.good()) {
+                    ReadConfigStream(include_config);
+                    printf("Included configuration file %s\n", to_include.c_str());
```

Comment:
I don't understand why this is a print statement. It'll get printed every time bitcoind or bitcoin-qt is started, and each time bitcoin-cli is called.

I think just a normal log here is fine.

Reply:
Hm, you're right that it prints every time you call bitcoin-cli etc. That's ugly. I'll think of something.

Reply:
I'm removing this entirely. As LogPrintf it is NOP and as printf it is ugly. I think we can live without being told the include was successful. The error case will be spammy, but I think that's okay.

Reply:
LogPrintf gets stored and replayed when going to a logfile; we could fix it so the same thing happens with printtoconsole?

Reply:
@ajtowns Really? I thought I checked the log file and didn't see it. If it is replayed, I should add it back.

Reply:
I restored LogPrintf line.

Reply:
#13088 should add support replay for console logs as well as debug.log fwiw